### PR TITLE
feat(ld-stepper): add progress indicator component

### PR DESCRIPTION
**Pull request by renet**
# Description

Adds the `ld-stepper` component along with `ld-step`.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] unit tests
- [x] e2e tests
- [x] accessibility tests
- [x] tested manually

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: src/liquid/components/ld-stepper/ld-step/ld-step.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  /** Step may be skipped */
+  @Prop() optional = false
+  /** Step size */
+  @Prop() size?: 'sm' | 'lg'
+  /** Step was skipped */
+  @Prop() skipped = false
+  /** Vertical layout */
+  @Prop() vertical = false
+
+  @Event() ldstepselected: EventEmitter<SelectedDetail>
```
Comment (borisdiakur) [inline src/liquid/components/ld-stepper/ld-step/ld-step.tsx:72]: Please document the event.

File: src/liquid/components/ld-stepper/ld-step/ld-step.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  componentDidLoad() {
+    if (this.current) {
+      this.handleClick()
+    }
+  }
+
+  render() {
+    const Component = this.href ? 'a' : 'button'
```
Comment (borisdiakur) [inline src/liquid/components/ld-stepper/ld-step/ld-step.tsx:96]: Maybe let's call it `FocusableElement` just like the ref?

File: src/liquid/components/ld-stepper/ld-step/ld-step.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            (this.done || this.icon) && 'ld-step--with-icon',
+            this.lastActive && 'ld-step--last-active',
+            this.next && 'ld-step--next',
+            this.optional && 'ld-step--optional',
+            this.size && `ld-step--${this.size}`,
+            this.skipped && 'ld-step--skipped',
+            this.vertical && 'ld-step--vertical',
+          ])}
+          part="li"
+          role="listitem"
```
Comment (borisdiakur) [inline src/liquid/components/ld-stepper/ld-step/ld-step.tsx:116]: `role` shouldn't be necessary here as the `li` has the role implicitly.

**Review by borisdiakur — CHANGES_REQUESTED** [review]

File: src/liquid/components/ld-stepper/ld-step/ld-step.tsx (hunk 3 above)
Comment (renet) [reply src/liquid/components/ld-stepper/ld-step/ld-step.tsx:116]: I've read [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listitem_role#best_practices) that if I use `list-style: none`, the role is required to make it a list item again. Leaving as is.

**Review by borisdiakur — APPROVED** [review]

**Pull request merged**